src/main/kotlin/sidechain/ChainListener.kt

```diff
@@ -1,11 +1,12 @@
 package sidechain
 
 import com.github.kittinunf.result.Result
+import java.io.Closeable
 
 /**
  * Class listens for new [Block] in side block chain
  */
-interface ChainListener<Block> {
+interface ChainListener<Block> : Closeable {
```

Comment:
seems like we have to close listeners in main code too. not tests only.

Reply:
And for other tests too.

Reply:
There is a separate task for that. I think next sprint might be dedicated to all these kind of stuff like graceful shutdown